pkg/controllers/sync/controller.go

```diff
@@ -1104,58 +1111,120 @@ func (s *SyncController) reconcileClusterForCascadingDeletion(
 		return worker.StatusAllOK
 	}
 
-	// cascading-delete is enabled, wait for member objects to be deleted
+	if !clusterutil.IsClusterReady(&cluster.Status) && cascadingdeletion.IsCascadingDeleteEnabled(cluster) {
```

Comment:
Is there a need to check for `cascadingdeletion.IsCascadingDeleteEnabled(cluster)` here?

Reply:
The check here is unnecessary and has been removed.